llvm/lib/Target/AArch64/AArch64MIPeepholeOpt.cpp

```diff
@@ -125,8 +125,14 @@ struct AArch64MIPeepholeOpt : public MachineFunctionPass {
   template <typename T>
   bool visitADDSSUBS(OpcodePair PosOpcs, OpcodePair NegOpcs, MachineInstr &MI);
 
+  // Strategy used to split logical immediate bitmasks.
+  enum class SplitStrategy {
+    Intersect,
+    Disjoint,
+  };
   template <typename T>
-  bool visitAND(unsigned Opc, MachineInstr &MI, unsigned OtherOpc = 0);
+  bool trySplitLogicalImm(unsigned Opc, MachineInstr &MI,
+                          SplitStrategy Strategy, unsigned OtherOpc = 0);
```

Comment:
I'm happy to revert this refactoring and keep separate visitAND and visitEOR/ORR methods if that's preferable.

Reply:
Thanks, I'll do that just now (in hindsight, I should have done that from the start).